### PR TITLE
Add flows picker to secondary locations panel

### LINKED ISSUE #1760: Support multi-flows in the Visualization panel
### Description

Modify visualization panel and support displaying issues that have multiple flows.

- [ ] selecting a flow in the panel should update the selection service
- [ ] the panel should reflect the selected flow in the selection service

Example for rule with multiple flows: TBD

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**
Fixes #1760.
Styling similar to the combo boxes in the error list.

image

### CODE REVIEW

File: src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml [hunk 1]
```diff
@@ -238,6 +239,162 @@
                 </DataTrigger>
             </Style.Triggers>
         </Style>
+
+        <!-- Flows Combo-box styles -->
```
Comment (rita-g-sonarsource) [inline src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml:243]: As I've said earlier, I don't know any other way to change the styles without overriding the control template and redeclaring all the elements in it. From all the search that I did, this is the way to do it, and that's why I've also done it in the past for the list view.

If you disagree and think that this is not the way, please explain how you've done it.

Comment (duncanp-sonar) [reply src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml:243]: If we're changing the structure of the control then redefining the template makes sense; if all we are going is changing the colours then it feels wrong, but I don't know of an alternative that works in VS.

File: src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml [hunk 2]
```diff
@@ -238,6 +239,162 @@
...
+
+        <ControlTemplate x:Key="ComboBoxToggleButton" TargetType="ToggleButton">
```
Comment (duncanp-sonar) [inline src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml:245]: Is there anything I need to look at specifically in this template, or is it just a straight copy of the default template with the colours changed?

Comment (rita-g-sonarsource) [reply src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml:245]: Almost a copy paste, with irrelevant parts removed (i.e. disabled/editing state). I added style triggers for colors and ComboBoxItem template + visibility trigger.
```
<Style.Triggers>
    <DataTrigger Binding="{Binding CurrentIssue.Flows.Count}" Value="1">
        <Setter Property="Visibility" Value="Collapsed"/>
    </DataTrigger>
</Style.Triggers>
```
https://docs.microsoft.com/en-us/dotnet/desktop/wpf/controls/combobox-styles-and-templates?view=netframeworkdesktop-4.8

Comment (duncanp-sonar) [reply src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml:245]: I think it's worth commenting what is boilerplate and what isn't. It's not obvious and there's too much of it to follow easily.

I'd probably have put all of this in a separate resource dictionary in a separate file with appropriate comments and merged the resource dictionaries.

File: src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml [hunk 3]
```diff
@@ -308,8 +466,28 @@
...
+                      Grid.Row="1" 
+                      Style="{StaticResource FlowsComboBox}"
+                      ItemContainerStyle="{StaticResource FlowsComboBoxItemStyle}"
+                      ItemsSource="{Binding CurrentIssue.Flows}"
+                      SelectedItem="{Binding CurrentFlow, Mode=TwoWay}">
+                <ComboBox.ItemTemplate>
+                    <DataTemplate DataType="{x:Type models:IAnalysisIssueFlowVisualization}">
+                        <TextBlock Style="{StaticResource BaseNumberInShapeTextBlockStyle}">
+                            <TextBlock.Text>
+                                <MultiBinding StringFormat="{}Flow {0} ({1} locations)">
```
Comment (duncanp-sonar) [inline src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml:479]: What is the empty `{}` for?

**Review by duncanp-sonar — APPROVED** [review]
Looks good visually.

File: src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml (hunk 3 above)
Comment (rita-g-sonarsource) [reply src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml:479]: Leftovers from initial attempts - you need to add {} if there is no custom text, and it first it was just flow number.

**Pull request merged**